libraries/chain/asset_evaluator.cpp

```diff
@@ -45,12 +45,23 @@ namespace detail {
             "Asset extension reward percent must be less than 100% till HARDFORK_1774_TIME!");
       }
    }
+
+   // TODO review and remove code below and links to it after HARDFORK_BSIP_81_TIME
+   void check_asset_options_hf_bsip81(const fc::time_point_sec& block_time, const asset_options& options)
+   {
+      if (block_time < HARDFORK_BSIP_81_TIME) {
+         // Taker fees should be zero until activation of BSIP81
+         FC_ASSERT(!options.extensions.value.taker_fee_percent.valid() ||
+                   *options.extensions.value.taker_fee_percent == 0,
+                   "Taker fee must be 0% until HARDFORK_BSIP_81_TIME");
```

Comment:
It can't be valid. `0` is still invalid.

Reply:
I think that I understand your point.  You can confirm my understanding after I push the commits which incorporate your suggestions.

Reply:
@abitmore Please check whether ecdda07d36144de934ce151358d5c7c91e6fd05a addresses your point

Reply:
Perfect. Thanks.